src/libraries/Microsoft.Extensions.HostFactoryResolver/src/HostFactoryResolver.cs

```diff
@@ -208,6 +204,10 @@ public object CreateHost()
 
                     try
                     {
+                        // Set the async local to the instance of the HostingListener so we can filter events that
+                        // aren't scoped to this execution of the entry point.
+                        _currentListener.Value = this;
```

Comment:
Do you ever have to "unset" this?

Reply:
We don't have to because it's on another thread.